packages/flutter/lib/src/widgets/overlay.dart

```diff
@@ -945,3 +976,429 @@ class _RenderTheatre extends RenderBox with ContainerRenderObjectMixin<RenderBox
     ];
   }
 }
+
+/// A widget that renders its [overlayChild] on the [Overlay] specified by
+/// [overlayInfo].
+///
+/// The [overlayChild] will be rendered on the [Overlay] as if it was inserted
+/// using an [OverlayEntry], while it can depend on the same set of
+/// [InheritedWidget]s (such as [Theme]) that this widget can depend on.
+///
+/// This widget must be placed below the [Overlay] [overlayInfo] points to, and
+/// [overlayInfo] must not be null when [overlayChild] is not null.
+///
+/// {@tool dartpad}
+/// This example uses an [OverlayPortal] to build a tooltip that becomes visible
+/// when the user taps on the [child] widget. There's a [DefaultTextStyle] above
+/// the [OverlayPortal] controlling the [TextStyle] of both the [child] widget
+/// and the [overlayChild] widget, which isn't otherwise doable if the tooltip
+/// was added as an [OverlayEntry].
+///
+/// ** See code in examples/api/lib/widgets/overlay/overlay_portal.0.dart **
+/// {@end-tool}
+///
+/// See also:
+///
+///  * [OverlayEntry], the imperative API for inserting a widget to an [Overlay].
+class OverlayPortal extends RenderObjectWidget {
+
+  /// Creates a widget that renders the given `overlayChild` on the [Overlay]
+  /// specified by `overlayChild`.
+  const OverlayPortal({
+    super.key,
+    required this.overlayInfo,
+    required this.overlayChild,
+    required this.child,
+  }) : assert(overlayChild == null || overlayInfo != null);
+
+  /// A widget below this widget in the tree, but renders on the [Overlay]
+  /// given by [overlayInfo].
+  ///
+  /// The [overlayChild] widget, if not null, is placed below this widget in
+  /// the widget tree, allowing it to depend on [InheritedWidget] above it, and
+  /// be notified when the [InheritedWidget] changes.
+  ///
+  /// Unlike [child], [overlayChild] can visually extend outside the bounds
+  /// of this widget without being clipped, and receive hit-test events outside
+  /// of this widget's bounds, as long as it does not extend outside of the
+  /// [Overlay] it renders on.
+  final Widget? overlayChild;
+
+  /// A widget below this widget in the tree.
+  final Widget? child;
+
+  /// A location in a particular [Overlay].
+  ///
+  /// An [OverlayInfo] on the enclosing [Overlay] widget can be aquired using
+  /// [OverlayInfo.of].
+  ///
+  /// This parameter must not be null when [overlayChild] is not null.
+  final OverlayInfo? overlayInfo;
```

Comment:
Why can't the OverlayPortal do this itself?

Reply:
See https://github.com/flutter/flutter/pull/105335#discussion_r889726040

There're cases we want to target a specific `Overlay`.

Reply:
Yeah I think it's fine for there to be an `overlay` parameter that lets you specify which one to use, but having it default to the nearest one would make this API so much simpler.

Reply:
Renamed the default constructor since everyone hates the explicit `overlayInfo`.